src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -421,6 +421,19 @@ public void SaveChanges(IList<UnspentOutputs> unspentOutputs, IEnumerable<TxOut[
                         cacheItem.UnspentOutputs.Spend(unspent);
 
                         this.logger.LogTrace("Cache item after spend {0}:'{1}'.", nameof(cacheItem.UnspentOutputs), cacheItem.UnspentOutputs);
+
+                        if (this.rewindDataIndexCache != null)
```

Comment:
If this gets moved here, where do new outpoints get added into the `rewindDataIndexCache`?

Reply:
Rewind index only points to utxo that are spent and currently reside in the rewind store so we don't need to index new outputs